crates/noirc_frontend/src/lexer/token.rs

```diff
@@ -391,7 +393,17 @@ impl Attribute {
 
                 Attribute::Deprecated(name.trim_matches('"').to_string().into())
             }
-            ["test"] => Attribute::Test,
+            ["test"] => Attribute::Test(false),
+            ["test", name] => {
+                if name != &"should_fail" {
```

Comment:
why not `#[test(fails)]`?

Reply:
We'll also likely want to mirror rust in the future which would allow `#[test(fails = "failure message")]`

Reply:
`should_fail` reads clearer to me than `fails`. Thoughts @kevaundray, @vezenovm, @TomAFrench, others?

Reply:
I'd also prefer `should_fail` then we can do should_fail = "message here" similar to should_panic